design/cluster-scope-resource-filter.md

```diff
@@ -0,0 +1,502 @@
+``` bash
+velero backup create <backup-name>
+	--include-namespaced-resources=Deployment,Service,Endpoint,Pod,ReplicaSet
+```
+
+``` bash
+velero backup create <backup-name>
+	--include-resources=Deployment,Service,Endpoint,Pod,ReplicaSet
+```
+
+The following commands mean backup Deployment, Service, Endpoint, Pod and ReplicaSet resources in namespaces default and kube-system, and related cluster resources.
+``` bash
+velero backup create <backup-name>
+	--include-namespaces=default,kube-system
+	--include-namespaced-resources=Deployment,Service,Endpoint,Pod,ReplicaSet
+```
+
+``` bash
+velero backup create <backup-name>
+	--include-namespaces=default,kube-system
+	--include-resources=Deployment,Service,Endpoint,Pod,ReplicaSet
+```
+
+This means backup all resources except Ingress type resources in all namespaces, and related cluster resources.
+``` bash
+velero backup create <backup-name>
+	--exclude-namespaced-resources=Ingress
+	--include-namespaced-resources="*"
+```
+
+#### some namespaced resources + some additional cluster resources
+This means backup all resources in namespace in default, kube-system, and related cluster resources, plus all StorageClass cluster resources.
+``` bash
+velero backup create <backup-name>
+	--include-namespaces=default,kube-system
+	--include-namespaced-resources="*"
+	--include-cluster-scope-resources=StorageClass
+	--include-cluster-resources=true
+```
+
+This means backup Deployment, Service, Endpoint, Pod and ReplicaSet resources in all namespaces, and related cluster resources, plus all StorageClass cluster resources.
+``` bash
+velero backup create <backup-name>
+	--include-namespaced-resources=Deployment,Service,Endpoint,Pod,ReplicaSet
+	--include-cluster-scope-resources=StorageClass
+	--include-cluster-resources=true
+```
+
+This means backup Deployment, Service, Endpoint, Pod and ReplicaSet resources in default and kube-system namespaces, and related cluster resources, plus all StorageClass cluster resources.
+``` bash
+velero backup create <backup-name>
+	--include-namespaced-resources=Deployment,Service,Endpoint,Pod,ReplicaSet
+	--include-namespaces=default,kube-system
+	--include-cluster-scope-resources=StorageClass
+	--include-cluster-resources=true
+```
+
+This means backup Deployment, Service, Endpoint, Pod and ReplicaSet resources in default and kube-system namespaces, and related cluster resources, plus all cluster scope resources except StorageClass type resources.
+``` bash
+velero backup create <backup-name>
+	--include-namespaced-resources=Deployment,Service,Endpoint,Pod,ReplicaSet
+	--include-namespaces=default,kube-system
+	--exclude-cluster-scope-resources=StorageClass
+	--include-cluster-resources=true
+```
+
+#### some namespaced resources + all cluster resources
+This means backup all resources in namespace in default, kube-system, and all cluster resources.
+``` bash
+velero backup create <backup-name>
+	--include-namespaces=default,kube-system
+	--include-namespaced-resources=*
```

Comment:
is `--include-namespaced-resources=*` really needed?  Isn't this default behavior?

Reply:
It's not. With the old parameters, leaving include-resources blank is equivalent to `--include-resources=*`, but "blank means all" doesn't work with the new split parameters, as it means there's no way to include nothing for either namespaced or cluster-scoped. That's why the latest version of this design removes the notion of "include all by default" with `--include-namespaced-resources` and `--include-cluster-scope-resources`. For those, the `=*` is required in order to account for all use cases.